x/tradeshield/spec/bot.md

```diff
@@ -0,0 +1,29 @@
+# Off-Chain Logic for Order Executions
+
+## Check Spot Order Execution Criteria Logic
+
+### Requirements
+
+  - `SpotOrder` object contains both `order_price` and `order_type`.
+  - `market_price` can be retrieved from the oracle module.
+  - If `order_type` is `STOPLOSS` and `market_price` is less than or equal to `order_price` then trigger `executeOrders` function.
+  - If `order_type` is `LIMITSELL` and `market_price` is greater than or equal to `order_price` then trigger `executeOrders` function.
+  - If `order_type` is `LIMITBUY` and `market_price` is less than or equal to `order_price` then trigger `executeOrders` function.
+  - If `order_type` is `MARKETBUY` and regardless of both `market_price` and `order_price` values then trigger `executeOrders` function.
+
+## Check Perpetual Order Execution Criteria Logic
+
+### Requirements
+
+  - `PerpetualOrder` object contains both `trigger_price` and `perpetual_order_type`.
+  - `market_price` can be retrieved from the oracle module.
+  - If `perpetual_order_type` is `LIMITOPEN` and `market_price` is less than or equal to `trigger_price` then trigger `executeOrders` function.
+  - If `perpetual_order_type` is `LIMITCLOSE` and `market_price` is greater than or equal to `trigger_price` then trigger `executeOrders` function.
+  - If `perpetual_order_type` is `MARKETOPEN` and regardless of both `market_price` and `trigger_price` values then trigger `executeOrders` function.
```

Comment:
If we don't consider price in market open and close, can we execute it when a user creates it ?
In that case, we don't have to store them and will save a step for bot too

Reply:
good point, yes we can execute them right away, lets add a ticket to update that and I will update the doc to reflect that change